exporter/signalfxexporter/metricdata_to_signalfxv2.go

```diff
@@ -0,0 +1,403 @@
+		&distributionValue.Sum)
+
+	// 3. Each histogram bucket is converted to a cumulative counter called
+	// <basename>_bucket and will include a dimension called upper_bound that
+	// specifies the maximum value in that bucket. This metric specifies the
+	// number of events with a value that is less than or equal to the upper
+	// bound.
+	metricName := *sfxBaseDataPoint.Metric + "_bucket"
+	explicitBuckets := distributionValue.BucketOptions.GetExplicit()
+	if explicitBuckets == nil {
+		return sfxDataPoints, fmt.Errorf(
+			"unknown bucket options type for metric %q",
+			*sfxBaseDataPoint.Metric)
+	}
+	bounds := explicitBuckets.Bounds
+	sfxBounds := make([]*string, len(bounds)+1)
+	for i := 0; i < len(bounds); i++ {
+		sfxBounds[i] = float64ToDimValue(bounds[i])
+	}
+	sfxBounds[len(sfxBounds)-1] = infinityBoundSFxDimValue
+
+	for i, bucket := range distributionValue.Buckets {
+
+		// Adding the "upper_bound" dimension.
+		bucketDimensions := make([]*sfxpb.Dimension, len(sfxBaseDataPoint.Dimensions)+1)
+		copy(bucketDimensions, sfxBaseDataPoint.Dimensions)
+
+		bucketDimensions[len(bucketDimensions)-1] = &sfxpb.Dimension{
+			Key:   &upperBoundDimensionKey,
+			Value: sfxBounds[i],
+		}
+
+		bucketDP := *sfxBaseDataPoint
+		bucketDP.Dimensions = bucketDimensions
+		bucketDP.Metric = &metricName
+		bucketDP.MetricType = bucketMetricType
+		count := bucket.Count
+		bucketDP.Value = &sfxpb.Datum{IntValue: &count}
+
+		sfxDataPoints = append(sfxDataPoints, &bucketDP)
+	}
+
+	return sfxDataPoints, nil
+}
+
+func appendSummaryValues(
+	sfxDataPoints []*sfxpb.DataPoint,
+	sfxBaseDataPoint *sfxpb.DataPoint,
+	summaryValue *metricspb.SummaryValue,
+) ([]*sfxpb.DataPoint, error) {
+
+	// Translating summary values per symmetrical recommendations to Prometheus:
+	// https://docs.signalfx.com/en/latest/integrations/agent/monitors/prometheus-exporter.html#overview
+
+	// 1. The total count gets converted to a cumulative counter called
+	// <basename>_count.
+	// 2. The total sum gets converted to a cumulative counter called <basename>
+	count := summaryValue.GetCount().GetValue()
+	sum := summaryValue.GetSum().GetValue()
+	sfxDataPoints = appendTotalAndSum(
+		sfxDataPoints,
+		sfxBaseDataPoint,
+		&count,
+		&sum)
+
+	// 3. Each quantile value is converted to a gauge called <basename>_quantile
+	// and will include a dimension called quantile that specifies the quantile.
+	percentiles := summaryValue.GetSnapshot().GetPercentileValues()
+	if percentiles == nil {
+		return sfxDataPoints, fmt.Errorf(
+			"unknown percentiles values for summary metric %q",
+			*sfxBaseDataPoint.Metric)
+	}
+	metricName := *sfxBaseDataPoint.Metric + "_quantile"
+	for _, quantile := range percentiles {
+
+		// Adding the "quantile" dimension.
+		quantileDimensions := make([]*sfxpb.Dimension, len(sfxBaseDataPoint.Dimensions)+1)
+		copy(quantileDimensions, sfxBaseDataPoint.Dimensions)
+
+		// TODO: What happens if a dimension "quantile" was already specified?
```

Comment:
I think the one that comes last would be accepted and the other discarded, but I'll check with the ingest team.

Reply:
Confirmed that last one does win.